File: js/initial-pose.js
// First, we create a Topic object with details of the topic's name and message type.
var initialposeTopic = new ROSLIB.Topic({
    ros : ros,
    name : 'initialpose',
    messageType : 'geometry_msgs/PoseWithCovarianceStamped'
});

function initialPose() {
    var initialposeMsg = new ROSLIB.Message({
        header: {
            frame_id: "map"
        },
        pose: {
            pose: {
                position: {
                    x: 1.0,
                    y: -3.0,
                    z: 0.0
                },
                orientation: {
                    x: 0.0,
                    y: 0.0,
                    z: 0.0,
                    w: 1.0
                }
            },
            covariance: [0.25, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.25, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.06853891945200942]
        }
    });

    initialposeTopic.publish(initialposeMsg);

    console.log("Sent initialpose.");
}

File: scripts/initial_pose.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String
from geometry_msgs.msg import PoseWithCovarianceStamped

def talker():
    pub = rospy.Publisher('initialpose', PoseWithCovarianceStamped, queue_size=10)
    rospy.init_node('initial_pose', anonymous=True)

    msg = PoseWithCovarianceStamped()

    msg.header.frame_id = 'map'
    msg.pose.pose.position.x = 1.0
    msg.pose.pose.position.y = -3.0
    msg.pose.pose.position.z = 0.0
    msg.pose.pose.orientation.x = 0.0
    msg.pose.pose.orientation.y = 0.0
    msg.pose.pose.orientation.z = 0.0
    msg.pose.pose.orientation.w = 1.0
    msg.pose.covariance = [0.25, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.25, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.06853891945200942]

    rospy.sleep(0.5)
    pub.publish(msg)

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass